In [1]:
import pandas as pd
import numpy as np

In [2]:
# to display 2 digits in pandas dataframe float numbers
pd.options.display.float_format = '{:.2f}'.format
# display all columns and rows from a dataframe
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [3]:
# List of file names
files = ['weekly_patterns_2018_sample.csv.zip', 'weekly_patterns_2019_sample.csv.zip',
         'weekly_patterns_2020_sample.csv.zip', 'weekly_patterns_2021_sample.csv.zip',
         'weekly_patterns_2022_sample.csv.zip']


total_obs = 0
for file in files:
    # Read the data
    data = pd.read_csv(file, compression='zip')
    total_obs += len(data)
print(total_obs)
data.dtypes



6462235


placekey               object
city                   object
region                 object
date_range_start       object
date_range_end         object
raw_visit_counts        int64
visits_by_day          object
safegraph_brand_ids    object
naics_code              int64
postal_code             int64
brands                 object
dtype: object

In [4]:
chilis_obs = 0
chilis_df = pd.DataFrame() #initializing an empty df

for file in files:
    # Read the data
    data = pd.read_csv(file, compression='zip')
    
    # Filter the data for Chili's Grill & Bar
    data = data.loc[data['brands'].isin(["Chili's Grill & Bar"])]
    
    # Concatenate the data
    chilis_df = pd.concat([chilis_df, data], ignore_index = True)
    
    chilis_obs = len(chilis_df)
    
print(chilis_obs)

56242


In [5]:
chilis_df.dtypes

placekey               object
city                   object
region                 object
date_range_start       object
date_range_end         object
raw_visit_counts        int64
visits_by_day          object
safegraph_brand_ids    object
naics_code              int64
postal_code             int64
brands                 object
dtype: object

In [6]:
chilis_df.head(3)

placekey            city region           date_range_start  \
0  222-222@5pr-4pg-9fz     Lees Summit     MO  2018-12-24T00:00:00-06:00   
1  223-222@8f2-sqt-kj9  Mount Pleasant     TX  2018-08-27T00:00:00-05:00   
2  223-222@5pv-mqc-t5f          Peoria     IL  2018-03-19T00:00:00-05:00   

              date_range_end  raw_visit_counts           visits_by_day  \
0  2018-12-31T00:00:00-06:00               261   [26,1,43,45,52,53,41]   
1  2018-09-03T00:00:00-05:00               138  [10,24,14,13,23,24,30]   
2  2018-03-26T00:00:00-05:00               111  [10,10,16,15,21,16,23]   

                         safegraph_brand_ids  naics_code  postal_code  \
0  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        64081   
1  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        75455   
2  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        61614   

                brands  
0  Chili's Grill & Bar  
1  Chili's Grill & Bar  
2  Chili's Grill & Bar

In [7]:
chilis_df.shape

(56242, 11)

In [8]:
chilis_df.to_csv('wide_sample.csv', index=False)

saving wide data to csv

In [9]:
chilis_df["placekey"].value_counts()

placekey
223-222@5z5-3vw-y35    262
222-222@8gq-946-r49    262
222-222@8gb-pwj-wc5    262
zzw-223@5z6-wdr-fmk    262
222-222@5pr-3d9-yvz    262
222-222@8ts-zyn-s5z    262
222-222@8gb-7fs-7h5    262
223-222@5s9-xg6-bhq    262
222-222@63j-8yv-jsq    262
222-222@8fy-85z-nqz    262
222-222@5z4-zhm-tn5    262
223-222@63g-jgm-vcq    262
223-222@8gb-cfk-45f    262
222-222@5yv-j2d-t5f    262
zzw-222@65y-xgq-jqf    262
222-222@62j-t3g-yy9    262
223-222@8sx-jgb-nh5    262
zzw-222@8fz-thn-gkz    262
222-222@63r-9k5-6c5    262
223-222@8t8-cqt-x3q    262
223-222@5qw-stn-4gk    262
223-222@5py-6jy-7t9    262
222-222@8fy-dtf-wc5    262
zzw-223@5pr-4zv-tsq    261
228-222@8g7-4qd-5vf    261
222-222@5qf-fy5-f9f    261
222-222@63q-5q6-3t9    261
222-222@5pr-4pg-9fz    261
223-222@62k-2jt-wzf    261
222-222@5pw-58y-68v    261
222-222@8ts-3n5-jvz    261
223-222@5qv-y75-8jv    261
223-222@8sz-tf5-7h5    261
222-222@8tn-zgq-ch5    261
223-222@8fb-j7w-f4v    261
zzw-224@5zb-wx4-kxq    261
223-222@5pw-5sq-8sq

In [10]:
chilis_df["placekey"].describe()

count                   56242
unique                    232
top       223-222@5z5-3vw-y35
freq                      262
Name: placekey, dtype: object

as we can see there are 232 different chilis in this data set. After a google search you find that there are about 1200 chilis in the US, so in the data we have just under 20% of the chilis. As discussed in class, this is because the data provided to us is a smaller, random sample of all of them, so it should still be representative of the country due to the nature of random sampling. 

In [11]:
from datetime import datetime, date, timedelta
chilis_df['date_range_start'] = chilis_df['date_range_start'].str[:10]
chilis_df['date_range_end'] = chilis_df['date_range_end'].str[:10]
chilis_df.head(5)

chilis_df['year'] = chilis_df['date_range_start'].str[:4]

In [12]:
chilis_df['date_range_start'] = pd.to_datetime(chilis_df['date_range_start'])
chilis_df['date_range_end'] =  pd.to_datetime(chilis_df['date_range_end'])
chilis_df.head(5)


placekey            city region date_range_start date_range_end  \
0  222-222@5pr-4pg-9fz     Lees Summit     MO       2018-12-24     2018-12-31   
1  223-222@8f2-sqt-kj9  Mount Pleasant     TX       2018-08-27     2018-09-03   
2  223-222@5pv-mqc-t5f          Peoria     IL       2018-03-19     2018-03-26   
3  zzw-222@62j-wsq-ysq           Dover     NH       2018-05-07     2018-05-14   
4  228-222@5qv-x6c-gzf          Euless     TX       2018-02-12     2018-02-19   

   raw_visit_counts           visits_by_day  \
0               261   [26,1,43,45,52,53,41]   
1               138  [10,24,14,13,23,24,30]   
2               111  [10,10,16,15,21,16,23]   
3               113   [13,15,18,9,26,19,13]   
4               175   [26,14,34,9,34,34,24]   

                         safegraph_brand_ids  naics_code  postal_code  \
0  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        64081   
1  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        75455   
2  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        61614   
3  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511         3820   
4  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        76039   

                brands  year  
0  Chili's Grill & Bar  2018  
1  Chili's Grill & Bar  2018  
2  Chili's Grill & Bar  2018  
3  Chili's Grill & Bar  2018  
4  Chili's Grill & Bar  2018

In [13]:
#chilis_df.sort_values('date_range_start')

#chilis_df.head(3)

#a = sorted(chilis_df['date_range_end'])

#print(a)


I commented out the printing of it after seeing it because it was so long,
but I got the following results:
the final date is the week of december 26 2022 - january 2 2023 
the first date is the week of january 1 - 8 2018

In [14]:
# Split the variable visits_by_day into new variables
chilis_df[['dailyvisits0', 'dailyvisits1', 'dailyvisits2', 'dailyvisits3',
    'dailyvisits4', 'dailyvisits5', 'dailyvisits6']] = chilis_df['visits_by_day'].str.split(',', expand=True)
chilis_df['dailyvisits0'] = chilis_df['dailyvisits0'].str[1:3]
chilis_df['dailyvisits6'] = chilis_df['dailyvisits6'].str[0:2]

    

chilis_df.head(5)

placekey            city region date_range_start date_range_end  \
0  222-222@5pr-4pg-9fz     Lees Summit     MO       2018-12-24     2018-12-31   
1  223-222@8f2-sqt-kj9  Mount Pleasant     TX       2018-08-27     2018-09-03   
2  223-222@5pv-mqc-t5f          Peoria     IL       2018-03-19     2018-03-26   
3  zzw-222@62j-wsq-ysq           Dover     NH       2018-05-07     2018-05-14   
4  228-222@5qv-x6c-gzf          Euless     TX       2018-02-12     2018-02-19   

   raw_visit_counts           visits_by_day  \
0               261   [26,1,43,45,52,53,41]   
1               138  [10,24,14,13,23,24,30]   
2               111  [10,10,16,15,21,16,23]   
3               113   [13,15,18,9,26,19,13]   
4               175   [26,14,34,9,34,34,24]   

                         safegraph_brand_ids  naics_code  postal_code  \
0  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        64081   
1  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        75455   
2  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        61614   
3  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511         3820   
4  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        76039   

                brands  year dailyvisits0 dailyvisits1 dailyvisits2  \
0  Chili's Grill & Bar  2018           26            1           43   
1  Chili's Grill & Bar  2018           10           24           14   
2  Chili's Grill & Bar  2018           10           10           16   
3  Chili's Grill & Bar  2018           13           15           18   
4  Chili's Grill & Bar  2018           26           14           34   

  dailyvisits3 dailyvisits4 dailyvisits5 dailyvisits6  
0           45           52           53           41  
1           13           23           24           30  
2           15           21           16           23  
3            9           26           19           13  
4            9           34           34           24

In [15]:
chilis_df['dailyvisits0'] = chilis_df['dailyvisits0'].astype(float)
chilis_df['dailyvisits1'] = chilis_df['dailyvisits1'].astype(float)
chilis_df['dailyvisits2'] = chilis_df['dailyvisits2'].astype(float)
chilis_df['dailyvisits3'] = chilis_df['dailyvisits3'].astype(float)
chilis_df['dailyvisits4'] = chilis_df['dailyvisits4'].astype(float)
chilis_df['dailyvisits5'] = chilis_df['dailyvisits5'].astype(float)

chilis_df['dailyvisits6'] = chilis_df['dailyvisits6'].str.replace(']', '')

chilis_df['dailyvisits6'] = chilis_df['dailyvisits6'].astype(float)

In [16]:
chilis_df['id'] = chilis_df.index
chilis_df.head(5)

placekey            city region date_range_start date_range_end  \
0  222-222@5pr-4pg-9fz     Lees Summit     MO       2018-12-24     2018-12-31   
1  223-222@8f2-sqt-kj9  Mount Pleasant     TX       2018-08-27     2018-09-03   
2  223-222@5pv-mqc-t5f          Peoria     IL       2018-03-19     2018-03-26   
3  zzw-222@62j-wsq-ysq           Dover     NH       2018-05-07     2018-05-14   
4  228-222@5qv-x6c-gzf          Euless     TX       2018-02-12     2018-02-19   

   raw_visit_counts           visits_by_day  \
0               261   [26,1,43,45,52,53,41]   
1               138  [10,24,14,13,23,24,30]   
2               111  [10,10,16,15,21,16,23]   
3               113   [13,15,18,9,26,19,13]   
4               175   [26,14,34,9,34,34,24]   

                         safegraph_brand_ids  naics_code  postal_code  \
0  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        64081   
1  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        75455   
2  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        61614   
3  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511         3820   
4  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        76039   

                brands  year  dailyvisits0  dailyvisits1  dailyvisits2  \
0  Chili's Grill & Bar  2018         26.00          1.00         43.00   
1  Chili's Grill & Bar  2018         10.00         24.00         14.00   
2  Chili's Grill & Bar  2018         10.00         10.00         16.00   
3  Chili's Grill & Bar  2018         13.00         15.00         18.00   
4  Chili's Grill & Bar  2018         26.00         14.00         34.00   

   dailyvisits3  dailyvisits4  dailyvisits5  dailyvisits6  id  
0         45.00         52.00         53.00         41.00   0  
1         13.00         23.00         24.00         30.00   1  
2         15.00         21.00         16.00         23.00   2  
3          9.00         26.00         19.00         13.00   3  
4          9.00         34.00         34.00         24.00   4

In [17]:


chilis_df.head(5)

placekey            city region date_range_start date_range_end  \
0  222-222@5pr-4pg-9fz     Lees Summit     MO       2018-12-24     2018-12-31   
1  223-222@8f2-sqt-kj9  Mount Pleasant     TX       2018-08-27     2018-09-03   
2  223-222@5pv-mqc-t5f          Peoria     IL       2018-03-19     2018-03-26   
3  zzw-222@62j-wsq-ysq           Dover     NH       2018-05-07     2018-05-14   
4  228-222@5qv-x6c-gzf          Euless     TX       2018-02-12     2018-02-19   

   raw_visit_counts           visits_by_day  \
0               261   [26,1,43,45,52,53,41]   
1               138  [10,24,14,13,23,24,30]   
2               111  [10,10,16,15,21,16,23]   
3               113   [13,15,18,9,26,19,13]   
4               175   [26,14,34,9,34,34,24]   

                         safegraph_brand_ids  naics_code  postal_code  \
0  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        64081   
1  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        75455   
2  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        61614   
3  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511         3820   
4  SG_BRAND_072efa4069c7217d82229845c60e6a0d      722511        76039   

                brands  year  dailyvisits0  dailyvisits1  dailyvisits2  \
0  Chili's Grill & Bar  2018         26.00          1.00         43.00   
1  Chili's Grill & Bar  2018         10.00         24.00         14.00   
2  Chili's Grill & Bar  2018         10.00         10.00         16.00   
3  Chili's Grill & Bar  2018         13.00         15.00         18.00   
4  Chili's Grill & Bar  2018         26.00         14.00         34.00   

   dailyvisits3  dailyvisits4  dailyvisits5  dailyvisits6  id  
0         45.00         52.00         53.00         41.00   0  
1         13.00         23.00         24.00         30.00   1  
2         15.00         21.00         16.00         23.00   2  
3          9.00         26.00         19.00         13.00   3  
4          9.00         34.00         34.00         24.00   4

In [18]:
chilis_df['index'] = chilis_df.index

In [19]:
long_chilis_df = pd.wide_to_long(chilis_df, ['dailyvisits'], i = 'index', j = 'day', sep = '')

long_chilis_df.head()

raw_visit_counts                        safegraph_brand_ids  \
index day                                                                
0     0                 261  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
1     0                 138  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
2     0                 111  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
3     0                 113  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
4     0                 175  SG_BRAND_072efa4069c7217d82229845c60e6a0d   

                      placekey           visits_by_day date_range_end  \
index day                                                               
0     0    222-222@5pr-4pg-9fz   [26,1,43,45,52,53,41]     2018-12-31   
1     0    223-222@8f2-sqt-kj9  [10,24,14,13,23,24,30]     2018-09-03   
2     0    223-222@5pv-mqc-t5f  [10,10,16,15,21,16,23]     2018-03-26   
3     0    zzw-222@62j-wsq-ysq   [13,15,18,9,26,19,13]     2018-05-14   
4     0    228-222@5qv-x6c-gzf   [26,14,34,9,34,34,24]     2018-02-19   

          date_range_start            city  year  naics_code  id  \
index day                                                          
0     0         2018-12-24     Lees Summit  2018      722511   0   
1     0         2018-08-27  Mount Pleasant  2018      722511   1   
2     0         2018-03-19          Peoria  2018      722511   2   
3     0         2018-05-07           Dover  2018      722511   3   
4     0         2018-02-12          Euless  2018      722511   4   

                        brands  postal_code region  dailyvisits  
index day                                                        
0     0    Chili's Grill & Bar        64081     MO        26.00  
1     0    Chili's Grill & Bar        75455     TX        10.00  
2     0    Chili's Grill & Bar        61614     IL        10.00  
3     0    Chili's Grill & Bar         3820     NH        13.00  
4     0    Chili's Grill & Bar        76039     TX        26.00

In [20]:


long_chilis_df.shape

long_chilis_df['dailyvisits'].describe()

count   393694.00
mean        26.41
std         27.09
min          0.00
25%         13.00
50%         22.00
75%         34.00
max        819.00
Name: dailyvisits, dtype: float64

as we can see, the variable dailyvisits has a mean of 26.41. so on average, 26.41 people visit a single chilis per day. howeverthere is certainly outliers considering the max is 819, and the min is 0. the min being 0 makes sense because if a certain chilis had to close for something like a holiday or other reason, it makes sense that no one went because it was closed. the reason for the max being so high however, could be due to some promotion ran by a certain chilis, or maybe a large party rented out a resturant inflating the number of visits un-naturally. I have no missing values, however a lot of times in economic data, missing values come from people declining to answer certain questions. That problem is not persent here however. 

outliers - specifically in this case since most outliers are much larger than the average - can really affect the mean, in this case inflating it. This could lead the business manager of a chilis to think more people are going to their resturants than what is actually happen, leading them to over-order perishaable food, or make other similar mistakes. The median however is unaffected by these outliers(unlike the mean), and could instead be used to make decisions that are unbiased towards the outliers and may paint a more accurate picture

In [21]:
long_chilis_df['dailyvisits'] = long_chilis_df['dailyvisits'].astype(int)

long_chilis_df.head(5) #the conversion to type int works because I had previously 
                        #turned all dailyvisitsi variables to float types. 

raw_visit_counts                        safegraph_brand_ids  \
index day                                                                
0     0                 261  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
1     0                 138  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
2     0                 111  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
3     0                 113  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
4     0                 175  SG_BRAND_072efa4069c7217d82229845c60e6a0d   

                      placekey           visits_by_day date_range_end  \
index day                                                               
0     0    222-222@5pr-4pg-9fz   [26,1,43,45,52,53,41]     2018-12-31   
1     0    223-222@8f2-sqt-kj9  [10,24,14,13,23,24,30]     2018-09-03   
2     0    223-222@5pv-mqc-t5f  [10,10,16,15,21,16,23]     2018-03-26   
3     0    zzw-222@62j-wsq-ysq   [13,15,18,9,26,19,13]     2018-05-14   
4     0    228-222@5qv-x6c-gzf   [26,14,34,9,34,34,24]     2018-02-19   

          date_range_start            city  year  naics_code  id  \
index day                                                          
0     0         2018-12-24     Lees Summit  2018      722511   0   
1     0         2018-08-27  Mount Pleasant  2018      722511   1   
2     0         2018-03-19          Peoria  2018      722511   2   
3     0         2018-05-07           Dover  2018      722511   3   
4     0         2018-02-12          Euless  2018      722511   4   

                        brands  postal_code region  dailyvisits  
index day                                                        
0     0    Chili's Grill & Bar        64081     MO           26  
1     0    Chili's Grill & Bar        75455     TX           10  
2     0    Chili's Grill & Bar        61614     IL           10  
3     0    Chili's Grill & Bar         3820     NH           13  
4     0    Chili's Grill & Bar        76039     TX           26

In [22]:
long_chilis_df.shape

(393694, 14)

In [23]:
agg = long_chilis_df.groupby(['year']).agg(
{'dailyvisits': 'mean' })

agg.head()

dailyvisits
year             
2018        24.14
2019        31.10
2020        23.20
2021        27.69
2022        25.95

In [24]:
print('number of observations for each year:')
long_chilis_df['year'].value_counts()

number of observations for each year:


year
2018    79576
2021    78834
2020    78610
2022    78428
2019    78246
Name: count, dtype: int64

In [25]:
long_chilis_df = long_chilis_df.reset_index()

long_chilis_df['date'] = long_chilis_df['date_range_start'] + pd.to_timedelta(long_chilis_df['day'], unit='D')

long_chilis_df['dayoftheweek'] = long_chilis_df['date'].dt.day_name()



In [26]:
long_chilis_df['dayoftheweek'].value_counts()

dayoftheweek
Monday       56242
Tuesday      56242
Wednesday    56242
Thursday     56242
Friday       56242
Saturday     56242
Sunday       56242
Name: count, dtype: int64

dayofweek variable makes sense.

In [27]:
threshold = 34 #I chose this threshold because 25% of the data will be greater than it: 
                #this gets the top quartile

long_chilis_df['manyvisits'] = np.where(long_chilis_df['dailyvisits'] > threshold, 1, 0)

long_chilis_df['manyvisits'].value_counts()

manyvisits
0    298515
1     95179
Name: count, dtype: int64

In [28]:
long_chilis_df.head()

index  day  raw_visit_counts                        safegraph_brand_ids  \
0      0    0               261  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
1      1    0               138  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
2      2    0               111  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
3      3    0               113  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
4      4    0               175  SG_BRAND_072efa4069c7217d82229845c60e6a0d   

              placekey           visits_by_day date_range_end  \
0  222-222@5pr-4pg-9fz   [26,1,43,45,52,53,41]     2018-12-31   
1  223-222@8f2-sqt-kj9  [10,24,14,13,23,24,30]     2018-09-03   
2  223-222@5pv-mqc-t5f  [10,10,16,15,21,16,23]     2018-03-26   
3  zzw-222@62j-wsq-ysq   [13,15,18,9,26,19,13]     2018-05-14   
4  228-222@5qv-x6c-gzf   [26,14,34,9,34,34,24]     2018-02-19   

  date_range_start            city  year  naics_code  id               brands  \
0       2018-12-24     Lees Summit  2018      722511   0  Chili's Grill & Bar   
1       2018-08-27  Mount Pleasant  2018      722511   1  Chili's Grill & Bar   
2       2018-03-19          Peoria  2018      722511   2  Chili's Grill & Bar   
3       2018-05-07           Dover  2018      722511   3  Chili's Grill & Bar   
4       2018-02-12          Euless  2018      722511   4  Chili's Grill & Bar   

   postal_code region  dailyvisits       date dayoftheweek  manyvisits  
0        64081     MO           26 2018-12-24       Monday           0  
1        75455     TX           10 2018-08-27       Monday           0  
2        61614     IL           10 2018-03-19       Monday           0  
3         3820     NH           13 2018-05-07       Monday           0  
4        76039     TX           26 2018-02-12       Monday           0

In [29]:
long_chilis_df['day'].value_counts()

day
0    56242
1    56242
2    56242
3    56242
4    56242
5    56242
6    56242
Name: count, dtype: int64

In [30]:
table = pd.crosstab(long_chilis_df['region'] , long_chilis_df['manyvisits'] )

print(table)

manyvisits      0      1
region                  
AL            703   1124
AR           1408   4073
AZ          13849   2601
CA          36555   1917
CO           8440    569
CT            404      2
DE           1511    316
FL          23846  14262
GA           7611   3407
HI           1735     92
IA           3151    503
ID           1759     68
IL          14265   4040
IN          12016   2607
KS           2375   1279
KY           6198   1110
LA           7018   3944
MA           8919    223
MD           1548    321
MI           1965   1213
MN           1625    202
MO           5763   5206
MS           1998   1656
MT           1825      2
NC           9401   5215
NH           1764     63
NJ           5941   1311
NM           3429    225
NV           3527    134
NY          16411   1824
OH           6552   2590
OK           6234   4728
PA           1589    238
RI           1818      9
TN           6194   1121
TX          51172  24981
UT           1861     29
VA           6880    778


The state with the highest number of manyvisits observations is Texas: 
This alligns with an online search as Texas is that state with the most chilis. 

In [31]:
table = pd.crosstab(long_chilis_df['region'] , long_chilis_df['manyvisits'] , normalize = 'index')*100
#same table as above but with percentages instead of total number counts
#Done to control for state size and see which states have best performing resturants
print(table)

manyvisits      0     1
region                 
AL          38.48 61.52
AR          25.69 74.31
AZ          84.19 15.81
CA          95.02  4.98
CO          93.68  6.32
CT          99.51  0.49
DE          82.70 17.30
FL          62.57 37.43
GA          69.08 30.92
HI          94.96  5.04
IA          86.23 13.77
ID          96.28  3.72
IL          77.93 22.07
IN          82.17 17.83
KS          65.00 35.00
KY          84.81 15.19
LA          64.02 35.98
MA          97.56  2.44
MD          82.83 17.17
MI          61.83 38.17
MN          88.94 11.06
MO          52.54 47.46
MS          54.68 45.32
MT          99.89  0.11
NC          64.32 35.68
NH          96.55  3.45
NJ          81.92 18.08
NM          93.84  6.16
NV          96.34  3.66
NY          90.00 10.00
OH          71.67 28.33
OK          56.87 43.13
PA          86.97 13.03
RI          99.51  0.49
TN          84.68 15.32
TX          67.20 32.80
UT          98.47  1.53
VA          89.84 10.16
VT         100.00  0.00
WA         100.0

The state with the highest percentage of manyvisit = 1  observations is Arkansas, followed by Alabama, so while there are less chilis in these states, the ones that are there seem to be doing very well. 

In [32]:
long_chilis_df['region'].value_counts()


region
TX    76153
CA    38472
FL    38108
IL    18305
NY    18235
AZ    16450
IN    14623
NC    14616
GA    11018
MO    10969
OK    10962
LA    10962
OH     9142
MA     9142
CO     9009
VA     7658
TN     7315
KY     7308
NJ     7252
WI     5488
AR     5481
NV     3661
NM     3654
KS     3654
IA     3654
MS     3654
MI     3178
UT     1890
MD     1869
VT     1834
WV     1827
MN     1827
AL     1827
MT     1827
DE     1827
NH     1827
HI     1827
ID     1827
PA     1827
RI     1827
WA     1302
CT      406
Name: count, dtype: int64

In [33]:
long_chilis_df['core_biz_area'] = np.where((long_chilis_df['region'] == 'AL')
                                           | (long_chilis_df['region'] =='AR')
                                           | (long_chilis_df['region'] =='FL') 
                                           | (long_chilis_df['region'] =='GA')
                                           |(long_chilis_df['region'] == 'KS') 
                                           | (long_chilis_df['region'] =='LA')
                                           |(long_chilis_df['region'] == 'MI') 
                                           |(long_chilis_df['region'] == 'MO') 
                                           |(long_chilis_df['region'] == 'MS') 
                                           | (long_chilis_df['region'] =='NC') 
                                           |(long_chilis_df['region'] =='OK') 
                                           |(long_chilis_df['region'] == 'TX') 
                                           |(long_chilis_df['region'] == 'WV'), 1, 0)

long_chilis_df.head()

#These are the states where >30% of the dailyvisit counts were instances of manyvisits.
#I wanted to control for state size,
#So I opted to use the percentage of manyvisits as my metric, and looking at the data, 
#30% was a natural cutoff from just looking at the data

index  day  raw_visit_counts                        safegraph_brand_ids  \
0      0    0               261  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
1      1    0               138  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
2      2    0               111  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
3      3    0               113  SG_BRAND_072efa4069c7217d82229845c60e6a0d   
4      4    0               175  SG_BRAND_072efa4069c7217d82229845c60e6a0d   

              placekey           visits_by_day date_range_end  \
0  222-222@5pr-4pg-9fz   [26,1,43,45,52,53,41]     2018-12-31   
1  223-222@8f2-sqt-kj9  [10,24,14,13,23,24,30]     2018-09-03   
2  223-222@5pv-mqc-t5f  [10,10,16,15,21,16,23]     2018-03-26   
3  zzw-222@62j-wsq-ysq   [13,15,18,9,26,19,13]     2018-05-14   
4  228-222@5qv-x6c-gzf   [26,14,34,9,34,34,24]     2018-02-19   

  date_range_start            city  year  naics_code  id               brands  \
0       2018-12-24     Lees Summit  2018      722511   0  Chili's Grill & Bar   
1       2018-08-27  Mount Pleasant  2018      722511   1  Chili's Grill & Bar   
2       2018-03-19          Peoria  2018      722511   2  Chili's Grill & Bar   
3       2018-05-07           Dover  2018      722511   3  Chili's Grill & Bar   
4       2018-02-12          Euless  2018      722511   4  Chili's Grill & Bar   

   postal_code region  dailyvisits       date dayoftheweek  manyvisits  \
0        64081     MO           26 2018-12-24       Monday           0   
1        75455     TX           10 2018-08-27       Monday           0   
2        61614     IL           10 2018-03-19       Monday           0   
3         3820     NH           13 2018-05-07       Monday           0   
4        76039     TX           26 2018-02-12       Monday           0   

   core_biz_area  
0              1  
1              1  
2              0  
3              0  
4              1

In [34]:
core_agg = long_chilis_df.groupby(['core_biz_area']).agg(
{'dailyvisits': 'mean' })

core_agg.head()

dailyvisits
core_biz_area             
0                    20.26
1                    32.83

There is about 1.5X the amount of daily visits in the core biz area compared to non core locations. 
this makes sense as the core area was defined as such areas with 30% or more daily visits being "manyvisits", so 
the core area should have a higher average by definition. If I just make the core biz area the number of chilis
in a state rather than the states with well performing chilis, this number could be different. I wanted the measurement
to be independent of state size so this would not happen, and we get the above relationship between core_biz_area & dv.